File: cinemaSpring/movieproject/src/main/java/com/aloha/movieproject/domain/UserAuth.java
package com.aloha.movieproject.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserAuth {
    private Long no;
    private String userId;
    private String auth;
}

File: App/cinema_flutter/lib/model/user_auth.dart

class UserAuth {
  int? no;
  String? userId;
  String? auth;

  UserAuth({
    this.no, 
    this.userId,
    this.auth
  });

  // Auth ✨ ➡ Map 🎁
  Map<String, dynamic> toMap() {
    return {
      'no': no,
      'userId' : userId,
      'auth' : auth
    };
  }
  // Map 🎁 ➡ Auth ✨
  factory UserAuth.fromMap(Map<String, dynamic> map) {
    return UserAuth(
      no: map['no'],
      userId: map['userId'],
      auth: map['auth']
    );
  }
} 